rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/config/RocketMQAutoConfiguration.java

```diff
@@ -66,7 +66,7 @@ public DefaultMQProducer defaultMQProducer(RocketMQProperties rocketMQProperties
 
     @Bean
     @ConditionalOnClass(name = "com.fasterxml.jackson.databind.ObjectMapper")
-    @ConditionalOnMissingBean(ObjectMapper.class)
+    @ConditionalOnMissingBean(name = "rocketMQMessageObjectMapper")
```

Comment:
I didn't get the context of the PR but I essentially [asked a similar question](https://github.com/apache/rocketmq-spring/pull/3#issuecomment-446640788). 

Why are you creating an `ObjectMapper` at all? Wouldn't it be better to reuse the one that Spring Boot auto-configures?

Reply:
@k631583871  would you like to try the latest code which merged @snicoll 's changes.

Reply:
@walking98 I don't know if you're asking me. The issue has non english description so I didn't get it.

Assuming you're asking me, what I am trying to ask is why do you need a dedicated `ObjectMapper`? If you do you can't create it like that otherwise you're going to switch off the one Spring Boot auto-configures by default depending on the ordering of the auto-configurations. If you could clarify why this exists in the first place, I could probably help and recommend something

Reply:
Thanks for the quick reply @snicoll . The issue is "if there is another bean ObjectMapper existing in the context, the starter can not startup"
I guess the bean is injected by other AutoConfig/Config...

Reply:
That's what I am trying to get at. You should order this auto-configuration to run _after_ `JacksonAutoConfiguration`. That way, you have the guarantee that an `ObjectMapper` will be provided by Spring Boot. 

If you like to create a different one internally with different settings, you can also do so using `Jackson2ObjectMapperBuilder` that is provided by Spring Boot (same auto-configuration).

If a user has switched off `JacksonAutoConfiguration` and this part of your starter does not work, that's totally fine IMO: users that apply such customizations to the auto-configurations are responsible to manage such cases manually.

In general, I don't encourage you to use qualifiers like that. I am happy to submit a PR that expresses in code what I've just explained. Let me know if you want me to do that.

Reply:
Okay, I would like to learn more from your new PR. please help on it :-)

Reply:
> I don't know if you're asking me. The issue has non english description so I didn't get it.

@snicoll  I have change the title to english :-)

Reply:
> I didn't get the context of the PR but I essentially [asked a similar question](https://github.com/apache/rocketmq-spring/pull/3#issuecomment-446640788).
> 
> Why are you creating an `ObjectMapper` at all? Wouldn't it be better to reuse the one that Spring Boot auto-configures?

我没有自己创建 ObjectMapper ，我引入了其他的架包，其中有一个架包就创建了 ObjectMapper